apps/storefront-e2e/src/support/page-objects/checkout.page.ts

```diff
@@ -56,4 +57,35 @@ export class CheckoutPage extends AbstractSFPage {
       .its('response.body.data.attributes.orderReference')
       .as('createdOrderId');
   };
+
+  templateIsReady = () => {
+    //email input is ready
+    this.checkoutAsGuestForm
+      .getEmailInput()
+      .should('be.visible', { timeout: 10000 });
```

Comment:
I'm not quite sure about these `, { timeout: 10000 }` checks everywhere, they are not 100% reliable and may cause failures

I suggest waiting for API requests to resolve (I'm almost 100% sure our UI is API-data-driven) and only then check the visibility.

By default Cypress wait for 30 seconds for API requests to resolve -> this will be enough for us to wait and then do visibility checks that take 4seconds max (bug 4 seconds should be enough to process response body and render the component)

Reply:
https://docs.cypress.io/api/commands/intercept

API interceptors are easy to create and manage + you can wait for multiple requests to resolve and only than continue https://docs.cypress.io/api/commands/wait#Aliases

Reply:
We need to ensure that all elements are properly rendered on the page, and unfortunately this does not depend on api calls. On cypress side checkout page (not locally) takes ~10sec to completely build the checkout page with all forms and elements after all request. Some other pages also had some flakines, but wait for 3s is enough for that.  I can predict that due to limited resources of remote machines.